### ISSUE TITLE
Function check regression

### ISSUE THREAD

**Issue by ePirat** [mesonbuild/meson#5481]
There is a regression since meson 0.50.0, meson incorrectly detects not available functions as available since 0.50.0.

It seems this is caused by meson not passing some of the crossfile cflags that were passed before, it clearly lacks `c_args`:

### Meson 0.50.0
```
Running compile:
Working directory:  /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpw3x_eba0
Command line:  clang -L/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/lib /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpw3x_eba0/testfile.c -pipe -o /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpw3x_eba0/output.exe -Wl,-ios_version_min,9.0 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -O0 -D_POSIX_C_SOURCE=200112L -Wl,-no_weak_imports 

Code:
 #include <time.h>
#include <limits.h>

        #if defined __stub_clock_gettime || defined __stub___clock_gettime
        fail fail fail this function is not going to work
        #endif
        
int main() {
            void *a = (void*) &clock_gettime;
            long b = (long) a;
            return (int) b;
        }
Compiler stdout:
 
Compiler stderr:
``` 

### Meson 0.49.2
```
Running compile:
Working directory:  /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpsqqg32j7
Command line:  clang -I/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/include -L/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/lib /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpsqqg32j7/testfile.c -pipe -o /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpsqqg32j7/output.exe -mios-version-min=9.0 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -mios-version-min=9.0 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -g -O2 -Wl,-ios_version_min,9.0 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -O0 -std=c99 -D_POSIX_C_SOURCE=200112L -Wl,-no_weak_imports 

Code:
 #include <time.h>
#include <limits.h>

        #if defined __stub_clock_gettime || defined __stub___clock_gettime
        fail fail fail this function is not going to work
        #endif
        
int main() {
            void *a = (void*) &clock_gettime;
            long b = (long) a;
            return (int) b;
        }
Compiler stdout:
 
Compiler stderr:
 ld: weak import of symbol '_clock_gettime' not supported because of option: -no_weak_imports for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

My crossfile:

```
# Automatically generated by contrib makefile

[binaries]
c = 'clang'
cpp = 'clang++'
ar = 'ar'
strip = 'strip'
pkgconfig = 'pkg-config'
windres = 'aarch64-apple-darwin18.6.0-windres'

[properties]
needs_exe_wrapper = true
c_args = ['-mios-version-min=9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-mios-version-min=9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-I/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/include', '-g', '-O2', ]
c_link_args = ['-Wl,-ios_version_min,9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-L/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/lib', ]
cpp_args = ['-mios-version-min=9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-mios-version-min=9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-mios-version-min=9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-I/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/include', '-g', '-I/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/include', '-g', '-O2', ]
cpp_link_args = ['-Wl,-ios_version_min,9.0', '-arch', 'arm64', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk', '-L/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/lib', ]

[host_machine]
system = 'darwin'
cpu_family = 'aarch64'
cpu = 'aarch64'
endian = 'little'
```

**Comment by ePirat** [comment]
Bisect result:

```
d451a4bd97f827bb492fd0c0e357cb20b6056ed9 is the first bad commit
commit d451a4bd97f827bb492fd0c0e357cb20b6056ed9
Author: John Ericson <John.Ericson@Obsidian.Systems>
Date:   Tue Jan 8 21:59:50 2019 -0500

    Remove get_cross_extra_flags

    This is no longer needed, we just remove conditionals around it.

:040000 040000 d1c154516c0e4b368b726a117f63756537711628 e18e6eb2e2c41b16bfbc9361ea0797081e9d0aa8 M	mesonbuild
```

cc @Ericson2314

**Comment by Ericson2314** [comment]
@ePirat this sounds like issues I fixed in `0.50.1` and also `master`. Can you try there?

**Comment by ePirat** [comment]
Same issue on master:

```
Running compile:
Working directory:  /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpisko7k95
Command line:  clang -L/Users/epirat/Projects/VideoLAN/VLC/build-test/contrib/arm64-apple-iOS_9.0/lib /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpisko7k95/testfile.c -pipe -o /var/folders/rw/l97gj1cd6vd_wrf_zjcbycjc0000gn/T/tmpisko7k95/output.exe -Wl,-ios_version_min,9.0 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -O0 -std=c99 -D_POSIX_C_SOURCE=200112L -Wl,-no_weak_imports 

Code:
 #include <time.h>
#include <limits.h>

        #if defined __stub_clock_gettime || defined __stub___clock_gettime
        fail fail fail this function is not going to work
        #endif
        
int main() {
            void *a = (void*) &clock_gettime;
            long b = (long) a;
            return (int) b;
        }
Compiler stdout:
 
Compiler stderr:
 
Checking for function "clock_gettime" : YES 
```
